service/common/src/main/java/org/apache/polaris/service/catalog/io/ExceptionMappingFileIO.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.catalog.io;
+
+import java.net.UnknownHostException;
+import java.util.Map;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.polaris.core.exceptions.FileIOUnknownHostException;
+
+/** A {@link FileIO} implementation that wraps an existing FileIO and re-maps exceptions */
+public class ExceptionMappingFileIO implements FileIO {
```

Comment:
This approach LGTM, but is this class actually used?

Reply:
Adding this wrapper to `DefaultFileIOFactory` SGTM as well.

Reply:
Added it there

Reply:
Added it there